### PR TITLE
Fix lints in generated C++ code

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR fix lints in the generated C++, primarily in generated C++ header files.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -6,14 +6,16 @@ Checks:
...
 '
-# WarningsAsErrors: '*'
-HeaderFilterRegex: ''
+WarningsAsErrors:  '*'
+HeaderFilterRegex: '.*'
+ExcludeHeaderFilterRegex: 'include/.*'
 UseColor: true
 FormatStyle: 'file'
-ExtraArgs: ['-std=c++17']
+ExtraArgs: ['-std=c++20']
```
Comment (bernardnormier) [inline .clang-tidy:18]: We use C++20 types (std::span) for a test in C++20 mode; clang-tidy doesn't like -std=c++17 for that test.

File: cpp/src/slice2cpp/Gen.cpp [hunk 2]
```diff
@@ -1111,9 +1111,14 @@ Slice::Gen::ForwardDeclVisitor::visitEnum(const EnumPtr& p)
         H << "class ";
     }
     H << getDeprecatedAttribute(p) << fixKwd(p->name());
-    if (!unscoped && p->maxValue() <= 0xFF)
+    if (!unscoped)
```
Comment (bernardnormier) [inline cpp/src/slice2cpp/Gen.cpp:1114]: Unfortunately we have no test for unscoped: we should either test it or remove it.

File: cpp/src/slice2cpp/Gen.cpp [hunk 3]
```diff
@@ -1221,7 +1226,12 @@ Slice::Gen::ForwardDeclVisitor::visitConst(const ConstPtr& p)
       << typeToString(p->type(), false, scope, p->typeMetadata(), _useWstring) << " " << fixKwd(p->name()) << " "
       << getDeprecatedAttribute(p) << "= ";
     writeConstantValue(H, p->type(), p->valueType(), p->value(), _useWstring, p->typeMetadata(), scope);
-    H << ';' << sp;
+    H << ';';
+    if (!isConstexprType(p->type()))
```
Comment (pepone) [inline cpp/src/slice2cpp/Gen.cpp:1230]: I would add a comment here, not clear what is this about.

File: cpp/src/slice2cpp/Gen.cpp [hunk 4]
```diff
@@ -1543,6 +1553,10 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& p)
     }
     H << nl << deprecatedAttribute << retS << ' ' << fixKwd(name) << spar << paramsDecl << contextDecl << epar
       << " const;";
+    if (ret)
+    {
+        H << " // NOLINT:modernize-use-nodiscard";
```
Comment (bernardnormier) [inline cpp/src/slice2cpp/Gen.cpp:1558]: We don't generate a [[nodiscard]] for sync proxy functions returning something.

File: cpp/src/slice2cpp/Gen.cpp [hunk 5]
```diff
@@ -2743,12 +2758,20 @@ Slice::Gen::InterfaceVisitor::visitOperation(const OperationPtr& p)
...
     if (ret)
     {
         string typeS = inputTypeToString(ret, p->returnIsOptional(), interfaceScope, p->getMetadata(), _useWstring);
         responseParams.push_back(typeS + " " + returnValueParam);
         responseParamsDecl.push_back(typeS + " ret");
         responseParamsImplDecl.push_back(typeS + " ret");
+
+        if (!amd && !isConst.empty())
+        {
+            noDiscard = "[[nodiscard]] ";
```
Comment (pepone) [inline cpp/src/slice2cpp/Gen.cpp:2773]: What is the relation of nodiscard and const?

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

File: cpp/src/slice2cpp/Gen.cpp (hunk 5 above)
Comment (bernardnormier) [reply cpp/src/slice2cpp/Gen.cpp:2779]: Added comment.

File: cpp/src/slice2cpp/Gen.cpp (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/slice2cpp/Gen.cpp]: Added comment.

**Pull request merged**